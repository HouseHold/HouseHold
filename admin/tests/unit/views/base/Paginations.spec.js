import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Paginations from '@/views/base/Paginations'

Vue.use(CoreuiVue)

describe('Paginations.vue', () => {
  it('has a name', () => {
    expect(Paginations.name).toBe('Paginations')
  })
  it('has a created hook', () => {
    expect(typeof Paginations.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof Paginations.data).toMatch('function')
    const defaultData = Paginations.data()
    expect(defaultData.currentPage).toBe(3)
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Paginations)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Paginations', () => {
    const wrapper = shallowMount(Paginations)
    expect(wrapper.is(Paginations)).toBe(true)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Paginations)
    expect(wrapper.element).toMatchSnapshot()
  })
})
