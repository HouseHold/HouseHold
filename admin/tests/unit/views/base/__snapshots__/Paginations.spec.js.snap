// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginations.vue renders correctly 1`] = `
<div>
  <ccard-stub>
    <ccardheader-stub>
      <cicon-stub
        name="cil-justify-center"
      />
       
      <strong>
         Responsive bootstrap Pagination
      </strong>
       
      <div
        class="card-header-actions"
      >
        <a
          class="card-header-action"
          href="https://coreui.io/vue/docs/components/pagination"
          rel="noreferrer noopener"
          target="_blank"
        >
          <small
            class="text-muted"
          >
            docs
          </small>
        </a>
      </div>
    </ccardheader-stub>
     
    <ccardbody-stub>
      <h6>
        Default
      </h6>
       
      <cpagination-stub
        activepage="3"
        align="start"
        arrows="true"
        dots="true"
        doublearrows="true"
        limit="5"
        pages="10"
        responsive=""
      />
       
      <br />
       
      <h6>
        Small
      </h6>
       
      <cpagination-stub
        activepage="3"
        align="start"
        arrows="true"
        dots="true"
        doublearrows="true"
        limit="5"
        pages="10"
        size="sm"
      />
       
      <br />
       
      <div>
        <h6>
          Large
        </h6>
         
        <cpagination-stub
          activepage="3"
          align="start"
          arrows="true"
          dots="true"
          doublearrows="true"
          limit="5"
          pages="10"
          responsive=""
          size="lg"
        />
         
        <br />
      </div>
       
      <div>
        currentPage: 3
      </div>
    </ccardbody-stub>
  </ccard-stub>
   
  <ccard-stub>
    <ccardheader-stub>
      <cicon-stub
        name="cil-justify-center"
      />
       
      <strong>
         Pagination 
      </strong>
       
      <small>
        alignment
      </small>
    </ccardheader-stub>
     
    <ccardbody-stub>
      <h6>
        Left alignment (default)
      </h6>
       
      <cpagination-stub
        activepage="3"
        align="start"
        arrows="true"
        dots="true"
        doublearrows="true"
        limit="5"
        pages="10"
      />
       
      <br />
       
      <h6>
        Center alignment
      </h6>
       
      <cpagination-stub
        activepage="3"
        align="center"
        arrows="true"
        dots="true"
        doublearrows="true"
        limit="5"
        pages="10"
      />
       
      <br />
       
      <h6>
        Right (end) alignment
      </h6>
       
      <cpagination-stub
        activepage="3"
        align="end"
        arrows="true"
        dots="true"
        doublearrows="true"
        limit="5"
        pages="10"
      />
       
      <br />
       
      <div>
        currentPage: 3
      </div>
    </ccardbody-stub>
  </ccard-stub>
</div>
`;
